tests/src/rules/rename-default-import.js

```diff
@@ -0,0 +1,357 @@
+    }),
+    test({
+      code: `import PropTypes from 'prop-types';`,
+      options: [{'prop-types': 'PropTypes'}],
+    }),
+    test({
+      code: `import PropTypes, {Foo} from 'prop-types';`,
+      options: [{'prop-types': 'PropTypes'}],
+    }),
+    test({
+      code: `import {default as PropTypes} from 'prop-types';`,
+      options: [{'prop-types': 'PropTypes'}],
+    }),
+    test({
+      code: `import {Foo} from 'prop-types';`,
+      options: [{'prop-types': 'PropTypes'}],
+    }),
+    test({
+      code: `import {Foo, default as PropTypes} from 'prop-types';`,
+      options: [{'prop-types': 'PropTypes'}],
+    }),
+    test({
+      code: `import * as PropTypes from 'prop-types';`,
+      options: [{'prop-types': 'PropTypes'}],
+    }),
+    test({
+      code: `import * as propTypes from 'prop-types';`,
+      options: [{'prop-types': 'PropTypes'}],
+    }),
+    test({
+      code: `const PropTypes = require('prop-types');`,
+      options: [{'prop-types': 'PropTypes'}],
+    }),
+    test({
+      code: `const object = require('prop-types').object;`,
+      options: [{'prop-types': 'PropTypes'}],
+    }),
+    test({
+      code: `const PropTypes = require('prop-types');`,
+      options: [],
+    }),
+    test({
+      code: `require('prop-types');`,
+      options: [{'prop-types': 'PropTypes'}],
+    }),
+  ],
+  invalid: [
+    test({
+      code: `import propTypes from 'prop-types';`,
+      options: [{'prop-types': 'PropTypes'}],
+      output: `import PropTypes from 'prop-types';`,
+      errors: [{
+        ruleId: 'rename-default-import',
+        message: `Default import from 'prop-types' should be aliased to PropTypes, not propTypes`,
+      }],
+    }),
+    test({
+      code: `const propTypes = require('prop-types');`,
+      options: [{'prop-types': 'PropTypes'}],
+      output: `const PropTypes = require('prop-types');`,
+      errors: [{
+        ruleId: 'rename-default-import',
+        message: `Default import from 'prop-types' should be aliased to PropTypes, not propTypes`,
+      }],
+    }),
+    test({
+      code: `import propTypes, {B} from 'prop-types';`,
+      options: [{'prop-types': 'PropTypes'}],
+      output: `import PropTypes, {B} from 'prop-types';`,
+      errors: [{
+        ruleId: 'rename-default-import',
+        message: `Default import from 'prop-types' should be aliased to PropTypes, not propTypes`,
+      }],
+    }),
+    test({
+      code: `import {default as propTypes} from 'prop-types';`,
+      options: [{'prop-types': 'PropTypes'}],
+      output: `import {default as PropTypes} from 'prop-types';`,
+      errors: [{
+        ruleId: 'rename-default-import',
+        message: `Default import from 'prop-types' should be aliased to PropTypes, not propTypes`,
+      }],
+    }),
+    test({
+      code: `import {default as propTypes, foo} from 'prop-types';`,
+      options: [{'prop-types': 'PropTypes'}],
+      output: `import {default as PropTypes, foo} from 'prop-types';`,
```

Comment:
i'm a little concerned with making this autofixable, if you won't be updating all the other places ijn the file where this variable is referenced (and it might not be safe to update it, if it's used in something like `const obj = { PropTypes };`).

Reply:
As far as I can tell this will update all the references apart from the exported identifiers. I am getting confused. Isn't it something that we agreed on?

Reply:
@ljharb can you please elaborate on this comment since I think it does what you said.

Reply:
Can we include test cases that have lots of references to the imported thing, so the autofix output can demonstrate that it updates too?